hedera-node/src/main/java/com/hedera/services/store/models/Token.java

```diff
@@ -234,6 +237,18 @@ public void wipe(OwnershipTracker ownershipTracker, TokenRelationship accountRel
 		setTotalSupply(newTotalSupply);
 	}
 
+    public void delete() {
+        validateTrue(hasAdminKey(), TOKEN_IS_IMMUTABLE);
+        validateFalse(isDeleted(), TOKEN_WAS_DELETED);
```

Comment:
When we do `TypedTokenStore.loadToken` we are executing `validateUsable` and there, we are checking whether the token was deleted. I think that we do not need to check it here as-well